### ISSUE TITLE
feature request: support PTP/USB

### ISSUE THREAD

**Issue by hkuno9000** [tengelmeier/mtp-tools#1]
I try to refine ptpcam codes for NIKON D5000/5500.
https://github.com/hkuno9000/ptpcam-fork/tree/bugfix-nikon-dc

To debug PTP_OC_NIKON_DirectCapture command,  I am using Writeshark2 with USBcap.
The URB_BULK in/out data is only HEX DUMP, that is so hard to parse for PTP.  
I look for Wireshark2 plugin of PTP/USB dissector, and found this project. but this is not PTP/USB, but PTP/IP.

Do you have a plan to support PTP/USB? or please suggest me to support it.

**Comment by tengelmeier** [comment]
Hi,

I have currently no plans to support USB as transport layer, but feel free to submit an patch / pull request. There is C version already in https://nikonhacker.com/wiki/Wireshark_PTP_Support.

You'll need -
- the original application to convert USB dumps to pcap
- port over the physical PTP-USB transport dissection to Lua
- make sure that the correct offset fields required by my chained MTP dissector are set. 

Regards,
             Thomas

**Comment by hkuno9000** [comment]
Thank. OK, I will try.

**Comment by tengelmeier** [comment]
If you need help, just ask.
I might have also some script to extract the vendor specific commands as JSON from ths SDK docs converted to HTML. I could be that the D5000 extraction relied on a textutil based .html, though... 

[EDIT] 
See feature/extract-nikon-commands

The workflow is: 
-> Download Nikon SDK docs 
-> convert the english PTP documentation word.doc to HTML with TextUtil 
-> use the Python scripts to extract the PTP/IP command description as JSON 
-> use the Python scripts to generate an extended dissector script for Wireshark.
